src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1370,7 +1370,10 @@ bool Creature::CanStartAttack(Unit const* who, bool force) const
     if (IsCivilian())
         return false;
 
-    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_NPC))
+    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_NPC) && who->GetTypeId() != TYPEID_PLAYER)  // flag is valid only for non player characters
+        return false;
+
+    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC) && who->GetTypeId() == TYPEID_PLAYER)  // immune to PC and target is a player, return false
         return false;
```

Comment:
`UNIT_FLAG_IMMUNE_TO_(N)PC` checks are done in `Creature::CanCreatureAttack`
we could delete this check here?